packages/web3torrent/src/library/paid-streaming-extension.ts

```diff
@@ -111,13 +117,36 @@ export abstract class PaidStreamingExtension implements Extension {
   onMessage(buffer: Buffer) {
     try {
       const jsonData = bencode.decode(buffer, undefined, undefined, 'utf8');
-      this.messageBus.emit(PaidStreamingExtensionEvents.NOTICE, jsonData);
+      this.messageHandler(jsonData);
     } catch (err) {
-      log('ERROR: decoding', err);
+      log('ERROR: onMessage decoding', err);
       return;
     }
   }
 
+  protected messageHandler({command, data}) {
+    switch (command) {
+      case PaidStreamingExtensionNotices.ACK:
+        return;
+      case PaidStreamingExtensionNotices.START:
+        log(`START received from ${this.peerAccount}`);
+        this.isBeingChoked = false;
+        this.wire.requests = [];
+        this.wire.unchoke();
```

Comment:
Could you explain why the leecher is choking/unchoking? Is it their way of reducing unecessary requests to the seeder?